### PR TITLE
[WIP] Add logger module

### PR DESCRIPTION
**Pull request by matthiasbeyer**

### CODE REVIEW

File: src/logger/module.c [hunk 1]
```diff
@@ -25,5 +25,105 @@
...
+    return 0;
+}
+
+void
+ws_log(
+    struct ws_logger_context* const ctx,
+    char* fmt,
+    ...
+) {
+    if (logger) {
```
Comment (neithernut) [inline src/logger/module.c]: Early return.

**Comment by matthiasbeyer** [comment]
@neithernut review? The last commit is one big bunch, I know. If you want cleanup, tell me.

**Comment by matthiasbeyer** [comment]
The `Fixup:` commits should be cleaned, though.

File: src/logger/module.c [hunk 2]
```diff
@@ -25,5 +25,95 @@
...
+ *
+ */
+
+int
+ws_logger_new(void)
+{
+    static bool is_used = false;
+
+    if (is_used) {
+        return -EBUSY;
```
Comment (neithernut) [inline src/logger/module.c]: return `0` in this case.

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
Ready for merging.

**Pull request merged**